sj201_revA/switch.py

```diff
@@ -0,0 +1,163 @@
+# Copyright 2020 Mycroft AI Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import RPi.GPIO as GPIO
+import time
+import threading
+ 
+class Switch:
+    """
+    Class to handle the Mark2 switches.
+    Note - a switch is an abstract concept 
+    which applies to buttons and switches.
+    The Mark2 actually has 4 different switches.
+    Three buttons (volume up, down and activate)
+    and a mute mic switch. All are read only
+    and interrupt driven. Also note switches are
+    pulled up so the active state is actually zero.
+    """
+    # GPIO pin numbers
+    _SW_ACTION = 22
+    _SW_VOL_UP = 23
+    _SW_VOL_DOWN = 24 
+    _SW_MUTE = 25
+
+    RED = (255,0,0)
+    GREEN = (0,255,0)
+    BLUE = (0,0,255)
+    BLACK = (0,0,0)
+    WHITE = (255,255,255)
```

Comment:
Wonder if these should be extracted out to an LED or colour class so they can be re-used across the board

Reply:
Across the board for sj201s. Other LEDs may have different characteristics. Did not want to get into having to map LED colors based on hardware characteristics. For example, a two state LED (on/off) would have a different underlying structure.

Reply:
So these are simply class level manifest constants. The objective is not to extend this out to the user in any way as I would not like to enumerate all color possibilities. If you would like I can search for an existing module which does so but it seems like a heavy lift to simply name a couple of constants. I suppose I could create a led_colors class and stick these in there if that would preferable.